gh-145036: Fix data race for list capacity (#145365)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2026-02-28-18-42-36.gh-issue-145036.70Kbfz.rst

> In free-threaded build, no longer change of race condition when calling :meth:`__sizeof__` on a :class:`list`

Reviewer: ```suggestion
In free-threaded build, no longer change of race condition when calling :meth:`!__sizeof__` on a :class:`list`
```

> In free-threaded build, no longer change of race condition when calling :meth:`!__sizeof__` on a :class:`list`

Reviewer: ```suggestion
In free-threaded build, fix race condition when calling :meth:`!__sizeof__` on a :class:`list`
```
